src/Compilers/CSharp/Portable/Symbols/Attributes/SourceAttributeData.cs

```diff
@@ -43,6 +45,7 @@ internal class SourceAttributeData : CSharpAttributeData
                 constructorArgumentsSourceIndices.Any() && constructorArgumentsSourceIndices.Length == constructorArguments.Length);
             Debug.Assert(attributeConstructor is object || hasErrors);
 
+            _compilation = compilation;
```

Comment:
Consider asserting that the `compilation` and `applicationNode` are not null

Reply:
> Consider asserting that the compilation and applicationNode are not null. This is also worth considering in some of the other impacted constructors

I will add the asserts in my next PR.